app/frontend/src/features/profile/ProfileTagInput.tsx

```diff
@@ -0,0 +1,229 @@
+    },
+  })
+);
+
+interface ProfileTagInputProps {
+  onChange: (_: any, value: string[]) => void;
+  value: string[];
+  options: string[];
+  label: string;
+  id: string;
+}
+
+export default function ProfileTagInput({
+  onChange,
+  value,
+  options,
+  label,
+  id,
+}: ProfileTagInputProps) {
+  const classes = useStyles();
+
+  const [open, setOpen] = useState<boolean>(false);
+  const anchorEl = useRef<null | HTMLButtonElement>(null);
+  const [valueState, setValueState] = useState<string[]>(value);
+  const [pendingValue, setPendingValue] = useState<string[]>([]);
+
+  useEffect(() => {
+    setValueState(value);
+  }, [value]);
+
+  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
+    setPendingValue(value);
+    setOpen(true);
+  };
+
+  const handleClose = (
+    event: React.ChangeEvent<{}>,
+    reason: AutocompleteCloseReason
+  ) => {
+    if (reason === "toggleInput") {
+      return;
+    }
+    onChange(null, pendingValue);
+    setOpen(false);
+  };
+
+  const handleRemove = (tag: string) => {
+    onChange(
+      null,
+      value.filter((v) => v !== tag)
+    );
+  };
+
+  const popperId = open ? id : undefined;
+
+  return (
+    <>
+      <ButtonBase
+        disableRipple
+        aria-describedby={popperId}
+        onClick={handleClick}
+        ref={anchorEl}
+        classes={{
+          root: classes.button,
+        }}
+      >
+        <span>{label}</span>
+        <ExpandMore />
+      </ButtonBase>
+      {valueState.map((tag) => (
+        <div key={tag} className={classes.tag}>
+          <IconButton
+            aria-label={`Remove ${tag}`}
+            edge="start"
+            onClick={() => handleRemove(tag)}
+          >
+            <CloseIcon fontSize="small" />
+          </IconButton>
+          <span className={classes.tagLabel}>{tag}</span>
+        </div>
+      ))}
+      <Popper
+        id={popperId}
+        open={open}
+        anchorEl={anchorEl.current}
+        placement="bottom-start"
+        className={classes.popper}
+      >
+        <div className={classes.header}>Press 'Enter' to add</div>
+        <Autocomplete
+          open
+          onClose={handleClose}
+          freeSolo
+          multiple
+          classes={{
+            paper: classes.paper,
+            option: classes.option,
+            popperDisablePortal: classes.popperDisablePortal,
+          }}
+          onChange={(_, newValue) => {
+            setPendingValue(newValue);
+          }}
+          value={pendingValue}
+          renderInput={(params) => (
+            <InputBase
+              ref={params.InputProps.ref}
+              inputProps={params.inputProps}
+              autoFocus
+              className={classes.inputBase}
+            />
+          )}
+          disableCloseOnSelect
+          disablePortal
+          options={options
+            .concat(value.filter((item) => options.indexOf(item) < 0))
+            .sort((a, b) => -b[0].localeCompare(a[0]))}
```

Comment:
```suggestion
          options={options}
```
All usage of this component so far seems to have this as an empty list, so you can just forward the value instead?

Reply:
Had a question about this. I imagine this will be filled with predefined options in the near future? I was also surprised to see that `freeSolo` was allowed. If we don't use `freeSolo`, this is not needed at all.

Reply:
Yes it will #409

Reply:
Okay will just leave this for now. It seems from the mid-fi that we may have use case for both predefined (languages, etc) and freeSolo (interests).